Sprint_4

Volard-QA opened the pull request.

Commit:
Added .gitignore

Commit:
Deleted manually created .gitignore

Commit:
Finished with needed tests in test.py

Commit:
Rewrite test_set_book_genre_set_genre_correctly in test.py

Commit:
Added .gitignore file

Commit:
Added .idea in .gitignore file

Commit:
Added last empty line in tests.py

Commit:
Added last empty line in tests.py

Commit:
Corrected test_get_books_for_children_not_get_books_with_genre_age_rating removed invalid method get_books_genre tests.py

Commit:
Corrected name test_add_new_book_not_add_same_book tests.py

Commit:
Добавлено описание тестов в файл README.md

Ayantea1 commented on tests.py:
```diff
@@ -18,7 +20,112 @@ def test_add_new_book_add_two_books(self):
+    #Проверяем добавление дублей книг
+    @pytest.mark.parametrize('book_name', ['Лабиринты Эхо', 'Вишневый сад', 'Цифровое искусство'])
+    def test_add_new_book_not_add_same_book(self, book_name):
+        collector = BooksCollector()
```
Можно лучше: чтобы не дублировать код в каждом тесте, можно создать фикстуру

Volard-QA replied:
Благодарю за комментарий!
Действительно оказалось интересно попрактиковаться в создании фикстуры, некоторые моменты стали понятнее, например, было не очевидно, что импортировать класс не обязательно в файле с основными тестами, если он импортирован в файле фикстуры, и более напрямую не вызывается в самих тестах. В новой версии файл с фикстурой добавил.

Ayantea1 commented on tests.py:
```diff
@@ -18,7 +20,112 @@ def test_add_new_book_add_two_books(self):
+        collector.add_new_book(book_2_name)
+        collector.set_book_genre(book_2_name, 'Фантастика')
+
+        assert collector.get_books_with_specific_genre('Фантастика') == [book_1_name, book_2_name]
```
Нужно исправить: сейчас тест не проверяет, что возвращаются только книги нужного жанра. Ты получил список всех добавленных книг

Volard-QA replied:
Тест дополнил!

Ayantea1 commented on tests.py:
```diff
@@ -18,7 +20,112 @@ def test_add_new_book_add_two_books(self):
+        collector.add_new_book(book_name)
+        collector.set_book_genre(book_name, genre)
+
+        assert collector.get_books_for_children() == []
```
Нужно исправить: этот тест показал, что метод может ничего не возвращать, а не то, что он работает как заявлено - возвращает только детские книги

Volard-QA replied:
Данный тест также переработал и дополнил.

Commit:
Добавлен файл conftest.py с фикстурой.

Скорректированы тесты test_get_books_with_specific_genre_return_valid_book_names
и test_get_books_for_children_not_get_books_with_genre_age_rating,
обнаружена и исправлена ошибка в тесте test_get_list_of_favorites_books_got_expected_books, почему-то отсутствовали методы add_new_book.

В файле tests.py убрана строка from main import BooksCollector, так как класс уже импортирован в отдельном файле фикстуры.

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,125 @@
+        assert collector.books_genre[book_name] == genre
+
+    #Проверяем соответствие выводимого функцией жанра установленному.
+    @pytest.mark.parametrize('book_name, expected_genre', [['Лабиринты Эхо', 'Фантастика']])
```
Можно лучше, здесь и далее: параметризация применяется чтобы запустить несколько раз один и тот же тест с разными параметрами,  а не чтобы их просто передать внутрь. Для одного набора данных она не имеет смысла. Можно объявить переменные внутри теста, количество кода это не поменяет
